Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1122,10 +1122,11 @@ void StyleComputer::collect_animation_into(DOM::Element& element, Optional<CSS::
             return keyframes.begin();
         }
         auto potential_match = keyframes.find_largest_not_above_iterator(key);
-        if (output_progress.value() >= 0) {
-            return --potential_match;
+        ++potential_match;
+        if (potential_match.is_end()) {
+            --potential_match;
         }
-        return potential_match;
+        return --potential_match;
```

Comment:
Just a small nit since this is probably not performance critical, but it feels wasteful to do this work for each interpolation step. We are traversing parts of the tree 3 to 4 times here.

You _could_ copy the iterator so you don't need that last traversal?
```cpp
        auto potential_match = keyframes.find_largest_not_above_iterator(key);
        auto probe = potential_match;
        ++probe;
        if (probe.is_end())
            --potential_match;
        return potential_match;
```

Reply:
yeah that seems reasonable